docs/BOOT_LOADER_SPECIFICATION.md

```diff
@@ -35,52 +35,56 @@ The target audience for this specification is:
 
 ## The boot partition
 
-Everything described below is located on a file system that will be called
-`$BOOT`. The boot loader or user-space programs reading the boot loader
-configuration should pick `$BOOT` according to the following rules:
+Everything described below is located on one or two partitions. The boot loader
+or user-space programs reading the boot loader configuration should locate the
+in the following manner:
 
 * On disks with an MBR partition table:
 
-  * A partition with the type ID of 0xEA shall be used as `$BOOT`.
+  * The boot partition — partition with the type ID of 0xEA — shall be used
+    for boot loader configuration and entries.
 
 * On disks with GPT (GUID Partition Table)
 
-  * If an Extended Boot Loader Partition (XBOOTLDR partition for short) — a
-    partition with GPT type GUID of `bc13c2ff-59e6-4262-a352-b275fd6f7172` —
-    is found, it should be used as `$BOOT`.
+  * The EFI System Partition (ESP for short) — a partition with GPT type GUID
+    of `c12a7328-f81f-11d2-ba4b-00a0c93ec93b` — should be used for boot loader
+    configuration and boot entries.
 
-  * Otherwise, the EFI System Partition (ESP for short) — a partition with
-    GPT type GUID of `c12a7328-f81f-11d2-ba4b-00a0c93ec93b` — should
-    be used as `$BOOT`.
+  * Optionally, an Extended Boot Loader Partition (XBOOTLDR partition for
+    short) — a partition with GPT type GUID of
+    `bc13c2ff-59e6-4262-a352-b275fd6f7172` — may be used as an additional
+    location for boot loader entries.
+
+In the text below, `$BOOT` will be used to refer to (the root of) the first of
+the two partitions (the boot partition on MBR disks and the ESP on GPT disks),
+and `$BOOT2` will be used to refer to (the root of) the optional second partition.
```

Comment:
`$XBOOT` or `$XBOOTLDR` would be better

Reply:
Maybe just use ESP, XBOOTLDR and $BOOT if it's either?

Reply:
There are following cases:
1. ESP or MBR /boot (for config)
2. ESP or MBR /boot and XBOOTLDR (for entries)

And I think in 2. it's better to explicitly say that both places are checked, then to imply that by using something like $BOOT to refer to either. So overall, $BOOT doesn't seem to be useful in the text at all. I think that especially with s/$BOOT2/XBOOTLDR/ the text will be much more comprehensible than before.